### PR TITLE
WebSocket 2

### PR DESCRIPTION
**Pull request by DedShotTM**
Continuation of https://github.com/LibraryofCode/discordts/pull/8

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,10 +14,14 @@
     "prepare": "npm run build"
   },
   "dependencies": {
-    "typescript": "^4.0.2"
+    "node-fetch": "^2.6.1",
```
Comment (bsian03) [inline package.json:17]: We're not using node fetch, please uninstall

File: src/gateway/WebSocket.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
+/* eslint-disable no-case-declarations */
+import WebSocket from 'ws';
+import fetch from 'node-fetch';
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import { GATEWAY_OPCODES, GATEWAY_CLOSE_CODES, EVENTS, Heartbeat, Payload } from './constants';
+
+export default async function Socket(token: string, intents: number) {
+  const res = await fetch('https://discord.com/api/v8/gateway/bot', {
```
Comment (bsian03) [inline src/gateway/WebSocket.ts:8]: Use inbuilt requester

File: src/gateway/constants.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
+// https://discord.com/developers/docs/topics/opcodes-and-status-codes#gateway-gateway-opcodes
```
Comment (bsian03) [inline src/gateway/constants.ts:1]: Opcodes and Close codes are in `/src/util/Constants.ts`

**Review by bsian03 — CHANGES_REQUESTED** [review]

**Comment by Khaaz** [comment]
why does this look the same as the already merged PR?

**Comment by bsian03** [comment]
Still wanted changes. I reset before the merge

File: src/gateway/constants.ts (hunk 3 above)
Comment (DedShotTM) [reply src/gateway/constants.ts]: Done

**Pull request merged**